src/Composer/Satis/Builder/ArchiveBuilder.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+/**
+ * This file is part of Satis.
+ *
+ * (c) Jordi Boggiano <j.boggiano@seld.be>
+ *     Nils Adermann <naderman@naderman.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace Composer\Satis\Builder;
+
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Helper\HelperSet;
+use Composer\Factory;
+use Composer\IO\ConsoleIO;
+use Composer\Package\PackageInterface;
+
+/**
+ * Builds the archives of the repository.
+ *
+ * @author James Hautot <james@rezo.net>
+ */
+class ArchiveBuilder extends Builder implements BuilderInterface
+{
+    /** @var InputInterface $input The input Interface. */
+    private $input;
+
+    /** @var HelperSet $helperSet A HelperSet instance. */
+    private $helperSet;
+
+    /**
+     * Builds the archives of the repository.
+     *
+     * @param array $packages List of packages to dump
+     */
+    public function dump(array $packages)
+    {
+        $helper = new ArchiveBuilderHelper($this->output, $this->config['archive']);
+
+        $directory = $helper->getDirectory($this->outputDir);
+
+        $this->output->writeln(sprintf("<info>Creating local downloads in '%s'</info>", $directory));
+
+        $format = isset($this->config['archive']['format']) ? $this->config['archive']['format'] : 'zip';
+        $endpoint = isset($this->config['archive']['prefix-url']) ? $this->config['archive']['prefix-url'] : $this->config['homepage'];
+
+        $includeArchiveChecksum = isset($this->config['archive']['checksum']) ? (bool) $this->config['archive']['checksum'] : true;
+
+        $composerConfig = Factory::createConfig();
+        $factory = new Factory();
+        $io = new ConsoleIO($this->input, $this->output, $this->helperSet);
+        $io->loadConfiguration($composerConfig);
+
+        /* @var \Composer\Downloader\DownloadManager $downloadManager */
+        $downloadManager = $factory->createDownloadManager($io, $composerConfig);
+
+        /* @var \Composer\Package\Archiver\ArchiveManager $archiveManager */
+        $archiveManager = $factory->createArchiveManager($composerConfig, $downloadManager);
+
+        $archiveManager->setOverwriteFiles(false);
+
+        shuffle($packages);
```

Comment:
The purpose of this PR is just refactoring, not changing the behavior, so I just copied the code :)

shuffling was proposed by @joshdifabio and merged by @naderman in #225